packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1718,6 +1718,14 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   // the text may position outside the visible region even when the text fits.
   bool get _hasVisualOverflow => _maxScrollExtent > 0 || _paintOffset != Offset.zero;
 
+  /// {@macro flutter.painting.textPainter.computeLineMetrics}
+  ///
+  /// Valid only after [layout].
+  List<ui.LineMetrics> computeLineMetrics() {
+    assert(!debugNeedsLayout);
+    return _textPainter.computeLineMetrics();
```

Comment:
This doesn't seem right:

1. `RenderEditable` may have a different paint origin from its `TextPainter` (because a `RenderEditable` can scroll).
2. In rare cases, `RenderEditable` may invalidate its `_textPainter`'s layout during intrinsic calculation.